aip/client-libraries/4234.md

```diff
@@ -0,0 +1,103 @@
+---
+aip:
+  id: 4234
+  scope: client-libraries
+  state: draft
+  created: 2019-06-28
+  updated: 2019-07-01
+permalink: /client-libraries/4234
+redirect_from:
+  - /4234
+---
+
+# Bazel protobuf targets
+
+When using Bazel to compile protobuf and generate message libraries and RPC service
+libraries, the naming choice for these targets has implications on derivative targets, and
+those target names have implications for per-language integrations. By applying a common
+approach to how individual `.proto` files are combined into targets, and to what those
+targets are named, clients can have a uniform experience from API to API.
+
+Some of the various language constraints are as follows:
+
+*   Java's `option java_outer_classname`: per the API style guide is the name of
+    the proto file, with "Proto" appended. Unless `option java_multiple_files` is true,
+    the file name defines the packaging of the types.
+*   Go's target names will define the import path for the result. However, Go's package
+    names have a more restrictive character set (e.g., no dashes), and in some cases, that
+    means that the import path and package name will disagree. This interferes with some
+    tooling.
+
+## Guidance
+
+### Targets
+
+In the following file name cases, if the service name or resource name is more than one
+word, they should be separated by underscores.
+
+#### Services and request/response messages
+
+Individual services along with their request and response messages **should** live in
+separate `proto_library` targets, named `{service}_service_proto`.
+
+**NOTE:** Services **should** not share request or response messages. Reuse of resources
```

Comment:
I generally advise against reuse of resources also, preferring to copy the structure.
For common messages, we should link to [AIP-213](https://aip.dev/213).